PilotDataPlatform/dataops
# PILOT-1996: Update resource operations to new repo structure

## Summary

- Refactored resource operation endpoints to follow new repo structure. This includes a base and dispatcher class for simplifying the copy and delete logic, along with respective dependencies.
- Updated copy/delete response to reflect `FileStatusSchema`.
- Removed unused code pertaining to resource operations.
- Updated test cases to reflect changes made to endpoint structure and logic.

Notes to reviewer:

- The structuring of tests to adhere to new structure will be improved in this ticket, respectively: [PILOT-1998](https://indocconsortium.atlassian.net/browse/PILOT-1998).
- Services that will need to be updated afterwords as a result of this PR are: `pipelines, pipelinewatch, auth, BFF-web, front-end`


## JIRA Issues

https://indocconsortium.atlassian.net/browse/PILOT-1996

## Type of Change


## Testing

Are there any new or updated tests to validate the changes?


## Test Directions

- Functionality of endpoints should remain the same. This can be further confirmed by requesting resource copy/delete in Portal.
